### PR TITLE
docs: show exported functions

### PR DESCRIPTION
**Pull request by DevVersion**
* Changes the Dgeni setup to properly display exported functions (e.g. from cdk/coercion, cdk/a11y, cdk/platform)

**NOTE**: A few descriptions for the coercion won't show up for now. See related fix for Dgeni-packages: 
 https://github.com/angular/dgeni-packages/pull/270

### CODE REVIEW

File: src/cdk/table/can-stick.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ export interface CanStick {
 }
 
 /**
+ * @docs-private
```
Comment (crisbeto) [inline src/cdk/table/can-stick.ts]: I think we put the `@docs-private` after the description everywhere else.

File: tools/dgeni/common/normalize-function-parameters.ts [hunk 2]
```diff
@@ -33,23 +36,24 @@ export function normalizeMethodParameters(method: NormalizedMethodMemberDoc) {
         parameterName = parameterName.replace('?', '');
       }
 
-      if (!method.params) {
-        method.params = [];
+      if (!doc.params) {
```
Comment (crisbeto) [inline tools/dgeni/common/normalize-function-parameters.ts]: nit: you can shorten this to `doc.params = doc.params || []`.

File: tools/dgeni/processors/component-grouper.ts [hunk 3]
```diff
@@ -103,6 +107,8 @@ export class ComponentGrouper implements Processor {
         group.additionalInterfaces.push(doc);
       } else if (doc.docType === 'type-alias') {
         group.additionalTypeAliases.push(doc);
+      } else if (doc.docType === 'function') {
```
Comment (crisbeto) [inline tools/dgeni/processors/component-grouper.ts]: This whole `else if` chain is getting pretty long. Consider using a switch on the `doc.docType`.

File: tools/dgeni/templates/componentGroup.template.html [hunk 4]
```diff
@@ -79,6 +79,20 @@ <h3 id="additional_interfaces" class="docs-header-link docs-api-h3">
     {% endfor %}
   {%- endif -%}
 
+  {%- if doc.additionalFunctions.length -%}
+    <h3 id="additional_functions" class="docs-header-link docs-api-h3">
```
Comment (crisbeto) [inline tools/dgeni/templates/componentGroup.template.html]: Should this be separated with a dash, rather than an underscore?

File: tools/dgeni/templates/componentGroup.template.html [hunk 5]
```diff
@@ -79,6 +79,20 @@ <h3 id="additional_interfaces" class="docs-header-link docs-api-h3">
...
+      <span header-link="additional_functions"></span>
+      Additional functions
+    </h3>
+    {% for item in doc.additionalFunctions %}
+      {#
+        Since the function docs are having a similar structure as method docs, we use
```
Comment (crisbeto) [inline tools/dgeni/templates/componentGroup.template.html]: are having -> have

File: src/cdk/table/can-stick.ts (hunk 1 above)
Comment (DevVersion) [reply src/cdk/table/can-stick.ts]: I see, thought about creating a rule that enforces this now. Do you think it would be worth having a rule that enforces this?

File: tools/dgeni/processors/component-grouper.ts (hunk 3 above)
Comment (DevVersion) [reply tools/dgeni/processors/component-grouper.ts]: I'd prefer keeping it like this. Since this is most likely the last `docType` and all the `breaks` don't make the code less or more readable IMO.

File: tools/dgeni/templates/componentGroup.template.html (hunk 4 above)
Comment (DevVersion) [reply tools/dgeni/templates/componentGroup.template.html]: Don't think so. I've just made it consistent with the other header links.

**Comment by DevVersion** [comment]
@crisbeto Done. Addressed most of your comments.

**Comment by jelbourn** [comment]
Is there anything aside from functions in `coercion` and `platform` being exposed?

**Comment by DevVersion** [comment]
@jelbourn Yeah, the `Platform` class for example. The Dgeni issue should be resolved as soon as possible. In the meanwhile, since the `coercion` isn't added to the doc items yet, it shouldn't be blocked on that issue.
